pkg/jobs/utils/helpers.go

```diff
@@ -559,3 +563,68 @@ func GetMonitorAttempts(jobType string, curator *clustercuratorv1.ClusterCurator
 
 	return monitorAttempts
 }
+
+// Copied from https://github.com/stolostron/library-go which is no longer supported
+// but this is a useful func for local testing as we don't need to build the image
+// LoadConfig loads the kubeconfig and returns a *rest.Config
+// url: The url of the server
+// kubeconfig: The path of the kubeconfig, if empty the KUBECONFIG environment variable will be used.
+// context: The context to use to search the *rest.Config in the kubeconfig file,
+// if empty the current-context will be used.
+// Search in the following order:
+// provided kubeconfig path, KUBECONFIG environment variable, in the cluster, in the user home directory.
+// If the context is not provided and the url is not provided, it returns a *rest.Config the for the current-context.
+// If the context is not provided but the url provided, it returns a *rest.Config for the server identified by the url.
+// If the context is provided, it returns a *rest.Config for the provided context.
+func LoadConfig(
+	url,
+	kubeconfig,
+	context string,
+) (*rest.Config, error) {
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+	klog.V(5).Infof("Kubeconfig path %s\n", kubeconfig)
+	// If we have an explicit indication of where the kubernetes config lives, read that.
+	if kubeconfig != "" {
+		return configFromFile(url, kubeconfig, context)
+	}
+	// If not, try the in-cluster config.
+	if c, err := rest.InClusterConfig(); err == nil {
```

Comment:
`rest.InClusterConfig()` Is this what we actually want to happen 99.99% of the time in the real world? 
If that's case, I feel like the stuffs above is not screaming "this is dev only!!! don't do this!!!" Like the "KUBECONFIG" env var variable is a generic enough that it might get use accidentally and we will be heading off the wrong logic route. 

I am just a bit worry about regression or some weird issues that will be hard to figure out in the future.

Reply:
Thanks for taking a look Mike! As for regressions, we've been using that same code for a long time until server foundation removed it (See https://github.com/stolostron/cluster-curator-controller/pull/264) so we should be ok. I'm just putting the code back.

But from a code quality perspective, if you think it will look confusing then I will remove it and just add it manually when I'm testing locally.